### PR TITLE
Add managed profiler HTTP routes to the assistant runtime

### PR DESCRIPTION
**Pull request by m-abboud**
## Summary
- Extract tar.gz archive creation and size-cap logic from log-export-routes into shared archive-utils
- Add authenticated profiler routes: list runs, get run detail with markdown summary, export as tar.gz, delete with budget recalculation
- Register profiler routes in http-server alongside existing operational routes
- Add comprehensive profiler-routes tests covering empty state, 404s, active-run rejection, export, and budget recalculation
- Update ARCHITECTURE.md with profiler directory contract and runtime endpoint documentation

Part of plan: managed-assistant-profiler-runtime.md (PR 3 of 3)

---

<a href="https://app.devin.ai/review/vellum-ai/vellum-assistant/pull/23299" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: assistant/src/runtime/routes/profiler-routes.ts [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+  });
+}
+
+/**
+ * GET /v1/profiler/runs/:runId — detail view with manifest + markdown summary.
+ */
+function handleGetRun(runId: string): Response {
+  const runDir = getProfilerRunDir(runId);
+  if (!existsSync(runDir)) {
+    return httpError("NOT_FOUND", `Profiler run '${runId}' not found`, 404);
```
[the remark this answers is not in the document]
Comment (m-abboud) [reply assistant/src/runtime/routes/profiler-routes.ts]: Addressed in direct push to admin-profiler-mode/pr-3-profiler-routes

**Comment by m-abboud** [comment]
@codex review this PR again — the previous issues have been fixed in commit 32d089c790634adffceb32f2fe15217a0c6888d0

**Comment by m-abboud** [comment]
@devin review this PR again — the previous issues have been fixed in commit 32d089c790634adffceb32f2fe15217a0c6888d0
